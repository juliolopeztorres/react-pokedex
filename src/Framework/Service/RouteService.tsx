import React from 'react';
import { hot } from "react-hot-loader/root";
import { Route, Switch } from 'react-router-dom';
import HomeView from '../View/HomeView';
import ServiceContainer from '../DependencyInjection/ServiceContainer';
import ServiceContainerInterface from '../DependencyInjection/ServiceContainerInterface';
import NavBar from "../View/Component/NavBar";
import DarkModeContainer from "../View/Component/DarkModeContainer";
import Footer from "../View/Component/Footer";
import NotFoundView from "../View/NotFoundView";

type routes = 'home';

const serviceContainer: ServiceContainerInterface = new ServiceContainer();

const RouteService = () => {
  return <DarkModeContainer>
    <NavBar/>
    <div className="grow">
      <Switch>
        <Route exact path={getRoute('home')}>
          <HomeView/>
        </Route>
        <NotFoundView/>
      </Switch>
    </div>
    <Footer/>
  </DarkModeContainer>;
}

export default hot(RouteService)

export function getRoute(path: routes): string {
  // generatePath('/'), // generatePath(path, parameters) -> Path string and parameters an object
  const routes = {
    home: {
      route: "/",
    },
  };

  return routes[path].route;
}
