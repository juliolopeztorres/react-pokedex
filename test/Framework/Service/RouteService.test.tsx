import React from 'react'
import { createRoot, Root } from "react-dom/client"
import { act } from "@testing-library/react"
import { BrowserRouter, MemoryRouter } from "react-router-dom";

jest.mock("../../../src/Framework/View/HomeView", () => {
  return () => <div>
    Home view
  </div>
})

jest.mock("../../../src/Framework/View/NotFound", () => {
  return () => <div>
    Not found
  </div>
})

import RouteService from "../../../src/Framework/Service/RouteService";

let root: Root
let container: HTMLElement
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div")
  document.body.appendChild(container)

  root = createRoot(container)
})

afterEach(() => {
  // cleanup on exiting
  act(() => root.unmount())

  container.remove()
})

it('should render home', () => {
  act(() => {
    root.render(<BrowserRouter>
      <RouteService/>
    </BrowserRouter>)
  })

  expect(container).toMatchSnapshot()
})

it('should render not found', () => {
  act(() => {
    root.render(<MemoryRouter initialEntries={['/whatever']}>
      <RouteService/>
    </MemoryRouter>)
  })

  expect(container).toMatchSnapshot()
})
