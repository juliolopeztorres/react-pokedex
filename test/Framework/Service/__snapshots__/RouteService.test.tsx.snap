// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render home 1`] = `
<div>
  <div>
    Home view
  </div>
</div>
`;

exports[`should render not found 1`] = `
<div>
  <div>
    Not found
  </div>
</div>
`;
